api_tutorials/bait_cnn/bait_performance_cnn.py

```diff
@@ -0,0 +1,605 @@
+    # summarize history for accuracy
+    plt.plot(history.history['categorical_accuracy'])
+    plt.plot(history.history['val_categorical_accuracy'])
+    plt.title('Accuracy: ' + title)
+    plt.ylabel('accuracy')
+    plt.xlabel('epoch')
+    plt.legend(['train', 'test'], loc='upper left')
+    plt.show()
+    # summarize history for loss
+    plt.plot(history.history['loss'])
+    plt.plot(history.history['val_loss'])
+    plt.title('Loss: ' + title)
+    plt.ylabel('loss')
+    plt.xlabel('epoch')
+    plt.legend(['train', 'test'], loc='upper left')
+    plt.savefig("./plot_history_" + title + ".jpg")
+
+
+def plot_metric_history(history, title):
+    # list all data in history
+    print("History keys: " + history.history.keys())
+
+    row = 0
+    col = 0
+    num_plots = len(history.history) / 2.0  # valid and train plot together
+    rows = 4
+    cols = int(math.ceil(num_plots / float(rows)))
+
+    f, axes = plt.subplots(rows, cols, sharex=True, figsize=(36, 24))
+
+    if cols > 1:
+        for k in history.history.keys():
+
+            if 'val' not in k:
+                axes[row, col].plot(history.history[k])
+                axes[row, col].plot(history.history['val_' + k])
+
+                axes[row, col].set_ylabel(str(k))
+                axes[row, col].legend(['train', 'valid'], loc='upper left')
+                axes[row, col].set_xlabel('epoch')
+
+                row += 1
+                if row == rows:
+                    row = 0
+                    col += 1
+                    if row * col >= rows * cols:
+                        break
+
+        axes[0, 1].set_title(title)
+    else:
+        for k in history.history.keys():
+
+            if 'val' not in k:
+                axes[row].plot(history.history[k])
+                axes[row].plot(history.history['val_' + k])
+
+                axes[row].set_ylabel(str(k))
+                axes[row].legend(['train', 'valid'], loc='upper left')
+                axes[row].set_xlabel('epoch')
+
+                row += 1
+        axes[0].set_title(title)
+
+    plt.savefig("./metric_history_" + title + ".jpg")
+
+
+def weight_path_from_args(args):
+    save_weight = './bait_performance_cnn_model'
+
+    ignore = ['inputs', 'labels', 'bed_file', 'reference_fasta']
+    for arg in vars(args):
+        if arg in ignore:
+            continue
+
+        attr = getattr(args, arg)
+
+        if os.path.isdir(str(attr)) or os.path.isfile(str(attr)):
+            continue
+
+        if os.path.isabs(str(attr)):
+            attr = os.path.splitext(os.path.basename(attr))[0]
+
+        save_weight += '__' + str(arg) + '_' + str(attr)
+
+    save_weight += '.hd5'
+    print('save weight path:', save_weight)
+
+    return save_weight
+
+
+def weight_path_to_title(wp):
+    return wp.split('/')[-1].replace('__', '-')
+
+#
+# # from gist: https://gist.github.com/kukuruza/03731dc494603ceab0c5
+# from math import sqrt
+#
+#
+# def put_kernels_on_grid(kernel, pad=1):
```

Comment:
Can this be uncommented? Seems like a helpful fxn...

Reply:
it might be, but I failed in getting it to work...adding it here in a comment so that folks have something to work with...